### ISSUE TITLE
a bug of TriangularMesh

### ISSUE THREAD

**Issue by OrtnerMichael** [magpylib/magpylib#615]
its not a bug but a problem ...

this output

![image](https://github.com/magpylib/magpylib/assets/47384012/71ea49bb-3dc4-43c6-8997-84721edc37d8)

![image](https://github.com/magpylib/magpylib/assets/47384012/542286f6-2c16-4985-b2ba-4d8f995d3b24)

is generated by the following code.

```python
import magpylib as magpy
import numpy as np
import matplotlib.pyplot as plt

top = magpy.misc.Triangle(
    magnetization=(0,0,1),
    vertices= ((-1,-1,1), (1,-1,1), (0,2,1)),
)
bottom = magpy.misc.Triangle(
    magnetization=(0,0,1),
    vertices= ((-1,-1,-1), (0,2,-1), (1,-1,-1)),
)
coll = magpy.Collection(top, bottom)

prism1 = magpy.magnet.TriangularMesh.from_triangles(
    magnetization=(0, 0, 1), # overrides triangles magnetization
    triangles=[top, bottom],
    validate_closed=False,  # disable check
    validate_connected=False,  # disable check
)

points = list(top.vertices) + list(bottom.vertices)
prism2 = magpy.magnet.TriangularMesh.from_ConvexHull(
    magnetization=(0, 0, 1), # overrides triangles magnetization
    points=points,
)

possis = np.linspace((-2,0,0), (2,0,0), 50)
sens = magpy.Sensor(possis)

magpy.show(sens, prism2)

b1 = coll.getB(possis)
plt.plot(b1[:,2], ls='', marker='.', color='g', label='Triangle Collection')

b2 = prism1.getB(possis)
plt.plot(b2[:,2], ls='', marker='o', mfc=(0,0,0,0), mec='r', label='open TriangularMesh')

b3 = prism2.getB(possis)
plt.plot(b3[:,2], ls='', color='lightblue', marker='x', label='closed TriangularMesh')

plt.gca().legend()
plt.show()
```

the "**Problem**" is that when a mesh is not closed, the inside-outside check of `TriangularMesh` can fail randomly and add (or not add) the magnetization.

**Comment by OrtnerMichael** [comment]
we should add a warning when someone uses getB with a TriangularMesh that is not closed

**Comment by OrtnerMichael** [comment]
fixed with #598
